index.html

```diff
@@ -31,9 +31,21 @@
 
         let libView = new LibraryEntryPoint.LibraryView(configuration);
 
-        libView.on("itemClicked", function(item) {
+        libView.on("itemClicked", function (item) {
             console.log(item);
         })
+
+        libView.on("searchStarted", function (searchView) {
+            let searchText = searchView.getSearchText();
+            console.log(searchText);
+
+            // call search function (from web component) to return json.
+            // If json available, pass the json to search view.
+            // Otherwise pass null to search view, and search view will 
+            // run search locally
+            let searchResultItems = null;
+            searchView.updateSearchResultListItems(searchResultItems);
+        })
```

Comment:
In my opinion, I think `function (searchView)` should be `function (searchText)`, a search algorithm should not need to know about the view in anyway. For this function, it should return the search result to the caller (which is `librarie.js`). So for this case, it will be:

```js
libView.on("searchStarted", function (searchText) {
    return null; // Indicates that library should do its own search
});
```